dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -1255,5 +1255,33 @@ HRESULT WINAPI SHQueryRecycleBinW(LPCWSTR pszRootPath, LPSHQUERYRBINFO pSHQueryR
     pSHQueryRBInfo->i64Size = 0;
     pSHQueryRBInfo->i64NumItems = 0;
 
+    CComPtr<IRecycleBin> spRecycleBin;
+    HRESULT hr = GetDefaultRecycleBin(pszRootPath, &spRecycleBin);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    CComPtr<IRecycleBinEnumList> spEnumList;
+    hr = spRecycleBin->EnumObjects(&spEnumList);
+    if (!SUCCEEDED(hr))
+        return hr;
+
+    while (TRUE)
+    {
+        CComPtr<IRecycleBinFile> spFile;
+        hr = spEnumList->Next(1, &spFile, NULL);
+        if (hr == S_FALSE)
+            return S_OK;
```

Comment:
Would not it make sense to break the loop here, rather than immediately returning? Because ` return S_OK;` at the bottom of the function will never be reached.

Reply:
Both work, something like this is really up to personal preference.